### PR TITLE
mocked posthog in testing and accept project key as param

### PR DESCRIPTION
**Pull request by Wxl19980214**
and the fixture to conftest.py

Added one more parameter to  log call and log_api: posthog project api key

I think this way each package now has their own data separated?

### CODE REVIEW

File: tests/telemetry/test_telemetry.py [hunk 1]
```diff
@@ -271,7 +273,7 @@ def test_stats_off(monkeypatch):
     mock = Mock()
     posthog_mock = Mock()
     mock.patch(telemetry, '_get_telemetry_info', (False, 'TestUID'))
-    telemetry.log_api("test_action", "ploomber", "0.14.0")
+    telemetry.log_api("test_action", "ploomber", "0.14.0", MOCK_API_KEY)
```
Comment (idomic) [inline tests/telemetry/test_telemetry.py:276]: Same, not the right place.

**Review by idomic — CHANGES_REQUESTED** [review]
Make sure the key is in the right function

File: tests/telemetry/test_telemetry.py (hunk 1 above)
Comment (Wxl19980214) [reply tests/telemetry/test_telemetry.py:276]: I am using @edublancas's suggestion. I will make a Telemetry class and put stuff there so we dont have to put all that in every telemetry call

**Comment by Wxl19980214** [comment]
@idomic @edublancas Is this a good solution? I think this saves some trouble reentering information all the time

**Pull request merged**